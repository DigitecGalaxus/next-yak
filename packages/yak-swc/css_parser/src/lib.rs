mod parse_css;
mod to_css;

pub use parse_css::*;
pub use to_css::*;