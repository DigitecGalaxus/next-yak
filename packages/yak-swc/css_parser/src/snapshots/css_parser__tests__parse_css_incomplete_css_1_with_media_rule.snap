---
source: src/lib.rs
expression: "(state, declarations)"
---
(
    ParserState {
        is_inside_string: None,
        is_inside_comment: false,
        is_inside_property_value: false,
        is_inside_at_rule: false,
        current_scopes: [
            CssScope {
                name: ".foo",
                scope_type: "selector",
            },
            CssScope {
                name: ".fancy",
                scope_type: "selector",
            },
            CssScope {
                name: "@media (max-width: 600px)",
                scope_type: "at-rule",
            },
            CssScope {
                name: ".baz",
                scope_type: "selector",
            },
        ],
        current_declaration: Declaration {
            property: "",
            value: "",
            scope: [],
            closed: false,
        },
    },
    [
        Declaration {
            property: "background",
            value: "url('https://example.com')",
            scope: [
                CssScope {
                    name: ".foo",
                    scope_type: "selector",
                },
                CssScope {
                    name: ".fancy",
                    scope_type: "selector",
                },
            ],
            closed: true,
        },
        Declaration {
            property: "color",
            value: "red",
            scope: [
                CssScope {
                    name: ".foo",
                    scope_type: "selector",
                },
                CssScope {
                    name: ".fancy",
                    scope_type: "selector",
                },
                CssScope {
                    name: "@media (max-width: 600px)",
                    scope_type: "at-rule",
                },
                CssScope {
                    name: ".baz",
                    scope_type: "selector",
                },
            ],
            closed: true,
        },
    ],
)
